### PR TITLE
agent: Hook Kotlin's compare method for integral types

### PR DESCRIPTION
**Pull request by kyakdan**
Kotlin's integral types have `compareTo` methods that are compiled to calls into one of two `compare` overloads from the `kotlin.jvm.internal.Intrinsics` class:
1. `public static int compare(long thisVal, long anotherVal)`
2. `public static int compare(int thisVal, int anotherVal)`

### CODE REVIEW

File: examples/BUILD.bazel [hunk 1]
```diff
@@ -373,6 +373,28 @@ java_fuzz_target_test(
     runtime_deps = [":ExampleKotlinFuzzTarget"],
 )
 
+kt_jvm_library(
+    name = "ExampleKotlinValueProfileFuzzeTarget",
```
Comment (fmeum) [inline examples/BUILD.bazel:377]:
```suggestion
    name = "ExampleKotlinValueProfileFuzzTarget",
```

File: examples/BUILD.bazel [hunk 2]
```diff
@@ -373,6 +373,28 @@ java_fuzz_target_test(
...
+java_fuzz_target_test(
+    name = "ExampleKotlinValueProfileFuzzer",
+    allowed_findings = [
+        "com.code_intelligence.jazzer.api.FuzzerSecurityIssueMedium",
+    ],
+    fuzzer_args = [
+        "-use_value_profile=1",
+    ],
+    target_class = "com.example.ExampleKotlinValueProfileFuzzer",
+    runtime_deps = [":ExampleKotlinValueProfileFuzzeTarget"],
```
Comment (fmeum) [inline examples/BUILD.bazel:395]:
```suggestion
    runtime_deps = [":ExampleKotlinValueProfileFuzzTarget"],
```

File: examples/src/main/java/com/example/ExampleKotlinValueProfileFuzzer.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+package com.example
+
+import com.code_intelligence.jazzer.api.FuzzedDataProvider
+import com.code_intelligence.jazzer.api.FuzzerSecurityIssueMedium
+
+object ExampleKotlinValueProfileFuzzer {
+
+    @JvmStatic
+    fun fuzzerTestOneInput(data: FuzzedDataProvider) {
+        if (data.consumeInt().compareTo(0x1122) != 0) {
```
Comment (fmeum) [inline examples/src/main/java/com/example/ExampleKotlinValueProfileFuzzer.kt:24]: We should probably make this a 32-bit comparison as 16 bit values plus null bytes are more easily guessed by chance.

**Review by fmeum — APPROVED** [review]

File: examples/BUILD.bazel (hunk 1 above)
Comment (kyakdan) [reply examples/BUILD.bazel]: Done.

File: examples/BUILD.bazel (hunk 2 above)
Comment (kyakdan) [reply examples/BUILD.bazel]: Done.

File: examples/src/main/java/com/example/ExampleKotlinValueProfileFuzzer.kt (hunk 3 above)
Comment (kyakdan) [reply examples/src/main/java/com/example/ExampleKotlinValueProfileFuzzer.kt]: Good point. Adjusted.

**Pull request merged**